core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.containerpool.docker
+
+import scala.concurrent.Future
+import spray.json.JsObject
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext
+import whisk.core.container.HttpUtils
+import whisk.core.entity.size._
+import scala.util.Success
+import scala.util.Failure
+import java.time.Instant
+import whisk.core.container.RunResult
+import whisk.core.container.Interval
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import whisk.common.TransactionId
+import whisk.common.Logging
+import whisk.common.LoggingMarkers
+import whisk.core.entity.ByteSize
+import java.nio.charset.StandardCharsets
+import whisk.http.Messages
+import whisk.core.entity.ActivationResponse
+import whisk.core.containerpool.BlackboxStartupError
+import whisk.core.containerpool.WhiskContainerStartupError
+import whisk.core.containerpool.InitializationError
+import whisk.core.containerpool.Container
+import whisk.core.invoker.ActionLogDriver
+
+object DockerContainer {
+    /**
+     * Creates a container running on a docker daemon.
+     *
+     * @param transid transaction creating the container
+     * @param image image to create the container from
+     * @param userProvidedImage whether the image is provided by the user
+     *     or is an OpenWhisk provided image
+     * @param memory memorylimit of the container
+     * @param cpuShares sharefactor for the container
+     * @param environment environment variables to set on the container
+     * @param network network to launch the container in
+     * @param name optional name for the container
+     * @return a container
```

Comment:
returns a future which either completes with a container or one of two specific failures.